### ISSUE TITLE
Where to find 'Direct' and 'Supervised' code of wesam?

### ISSUE THREAD

**Issue by ChunmengLiu1** [zhang-haojie/wesam#13]
Hi, excellent work!

I want to re-implement the results of 'Direct' and 'Supervised'. Where could I find the code? Thank you for your help!

**Comment by zhang-haojie** [comment]
'Direct' means inference with the origin SAM. It is line 190 in `adaptation.py`. 'Supervised' means GT masks as supervise, you can refer to [lightning-sam](https://github.com/luca-medeiros/lightning-sam)

**Comment by ChunmengLiu1** [comment]
Got it! Thank you for your quickly response!

**Comment by ChunmengLiu1** [comment]
Hi, I read the code of [lightning-sam](https://github.com/luca-medeiros/lightning-sam). Do you mean that lightning-sam used the ground truth mask when calculate loss? The gt_mask is instance segmentation ground truth?

For train.py, line 94-100:
`            for pred_mask, gt_mask, iou_prediction in zip(pred_masks, gt_masks, iou_predictions):
                batch_iou = calc_iou(pred_mask, gt_mask)
                loss_focal += focal_loss(pred_mask, gt_mask, num_masks)
                loss_dice += dice_loss(pred_mask, gt_mask, num_masks)
                loss_iou += F.mse_loss(iou_prediction, batch_iou, reduction='sum') / num_masks
            loss_total = 20. * loss_focal + loss_dice + loss_iou`

Thanks for your help!

**Comment by ChunmengLiu1** [comment]
Hi! 
Can you give me more information about "supervised"?
For now, I replace soft_mask with gt_mask, and just keep loss_focal, loss_dice and loss_iou. Can you give me some more advice? It's the right setting? Thank you for your help.

```
            for i, (pred_mask, gt_mask, iou_prediction) in enumerate(zip(pred_masks, gt_masks, iou_predictions)):
                loss_focal += focal_loss(pred_mask, gt_mask, num_masks)
                loss_dice += dice_loss(pred_mask, gt_mask, num_masks)
                batch_iou = calc_iou(pred_mask, gt_mask)
                loss_iou += F.mse_loss(iou_prediction, batch_iou, reduction='sum') / num_masks
            loss_total = 20. * loss_focal + loss_dice + loss_iou
            fabric.backward(loss_total)
```

**Comment by zhang-haojie** [comment]
> Hi, I read the code of [lightning-sam](https://github.com/luca-medeiros/lightning-sam). Do you mean that lightning-sam used the ground truth mask when calculate loss? The gt_mask is instance segmentation ground truth?
> 
> For train.py, line 94-100:
> 
> ```
>             for pred_mask, gt_mask, iou_prediction in zip(pred_masks, gt_masks, iou_predictions):
>                 batch_iou = calc_iou(pred_mask, gt_mask)
>                 loss_focal += focal_loss(pred_mask, gt_mask, num_masks)
>                 loss_dice += dice_loss(pred_mask, gt_mask, num_masks)
>                 loss_iou += F.mse_loss(iou_prediction, batch_iou, reduction='sum') / num_masks
> 
>             loss_total = 20. * loss_focal + loss_dice + loss_iou
> ```
> 
> Thanks for your help!

Yes

**Comment by ChunmengLiu1** [comment]
> > Hi! Can you give me more information about "supervised"? For now, I replace soft_mask with gt_mask, and just keep loss_focal, loss_dice and loss_iou. Can you give me some more advice? Is it the right setting? Are loss_contra and loss_anchor required? Thank you for your help.
> > ```
> >             for i, (pred_mask, gt_mask, iou_prediction) in enumerate(zip(pred_masks, gt_masks, iou_predictions)):
> >                 loss_focal += focal_loss(pred_mask, gt_mask, num_masks)
> >                 loss_dice += dice_loss(pred_mask, gt_mask, num_masks)
> >                 batch_iou = calc_iou(pred_mask, gt_mask)
> >                 loss_iou += F.mse_loss(iou_prediction, batch_iou, reduction='sum') / num_masks
> >             loss_total = 20. * loss_focal + loss_dice + loss_iou
> >             fabric.backward(loss_total)
> > ```
> 
> For `supervised`, It's just plain supervised training, there's no `student` or `teacher` model, and there's no anchor loss

Thanks!
